code/game/machinery/washing_machine.dm

```diff
@@ -26,286 +20,227 @@
+				icon_state = initial(H.icon_state)
+				item_color = wash_color
+				name = initial(H.name)
+				desc = "The colors are a bit dodgy."
+				break
+
+
+/obj/machinery/washing_machine/relaymove(mob/user)
+	container_resist(user)
+
+/obj/machinery/washing_machine/container_resist(mob/user)
+	if(!busy)
+		add_fingerprint(user)
+		open_machine()
+
 
 
 /obj/machinery/washing_machine/update_icon()
-	icon_state = "wm_[state]0"
+	overlays.Cut()
+	if(busy)
+		icon_state = "wm_running_[bloody_mess]"
+	else if(bloody_mess)
+		icon_state = "wm_[state_open]_blood"
+	else
+		var/full = contents.len ? 1 : 0
+		icon_state = "wm_[state_open]_[full]"
+	if(panel_open)
+		overlays += image(icon, icon_state = "wm_panel")
 
 /obj/machinery/washing_machine/attackby(obj/item/weapon/W, mob/user, params)
-	if(istype(W,/obj/item/toy/crayon) ||istype(W,/obj/item/weapon/stamp))
-		if( state in list(	1, 3, 6 ) )
-			if(!crayon)
-				if(!user.drop_item())
-					return
-				crayon = W
-				crayon.loc = src
-			else
-				..()
-		else
-			..()
+	if(default_deconstruction_screwdriver(user, null, null, W))
+		update_icon()
+		return
 	else if(istype(W,/obj/item/weapon/grab))
-		if(state == 1)
+		if(state_open)
 			var/obj/item/weapon/grab/G = W
 			if(iscorgi(G.affecting))
+				has_corgi = 1
 				G.affecting.loc = src
 				qdel(G)
-				state = 3
-		else
-			..()
-
-	else if(istype(W,/obj/item/stack/sheet/hairlesshide) || \
-		istype(W,/obj/item/clothing/under) || \
-		istype(W,/obj/item/clothing/mask) || \
-		istype(W,/obj/item/clothing/head) || \
-		istype(W,/obj/item/clothing/gloves) || \
-		istype(W,/obj/item/clothing/shoes) || \
-		istype(W,/obj/item/clothing/suit) || \
-		istype(W,/obj/item/weapon/bedsheet) || \
-		istype(W,/obj/item/weapon/paper))
-
-		//YES, it's hardcoded... saves a var/can_be_washed for every single clothing item.
-		if ( istype(W,/obj/item/clothing/suit/space ) )
-			user << "This item does not fit."
-			return
-		if ( istype(W,/obj/item/clothing/suit/syndicatefake ) )
-			user << "This item does not fit."
-			return
-//		if ( istype(W,/obj/item/clothing/suit/powered ) )
-//			user << "This item does not fit."
-//			return
-		if ( istype(W,/obj/item/clothing/suit/cyborg_suit ) )
-			user << "This item does not fit."
-			return
-		if ( istype(W,/obj/item/clothing/suit/bomb_suit ) )
-			user << "This item does not fit."
-			return
-		if ( istype(W,/obj/item/clothing/suit/armor ) )
-			user << "This item does not fit."
-			return
-		if ( istype(W,/obj/item/clothing/suit/armor ) )
-			user << "This item does not fit."
-			return
-		if ( istype(W,/obj/item/clothing/mask/gas ) )
-			user << "This item does not fit."
-			return
-		if ( istype(W,/obj/item/clothing/mask/cigarette ) )
-			user << "This item does not fit."
-			return
-		if ( istype(W,/obj/item/clothing/head/syndicatefake ) )
-			user << "This item does not fit."
-			return
-		if ( istype(W,/obj/item/clothing/head/helmet ) )
-			user << "This item does not fit."
-			return

```

Comment:
jesus fuckign christ isf this dude is so allergic to a new var on every clothes item the least he could have done was made these one check with a gorillion ||s
what the fuck how could he possibly have decided that this was the best solution